New package - Gate

glennpj opened the pull request:
This PR adds the Gate package as well as the ITK dependency.

glennpj committed:
New package - Gate

This PR adds the Gate package as well as the ITK dependency.

glennpj commented:
Note that I will be submitting patches upstream.

glennpj committed:
Fix flake 8 errors

adamjstewart reviewed (approved):
It might be a good idea to explicitly disable some of these CMake flags in else clauses.

glennpj committed:
Be more explicit with CMake options

Make sure Cmake values related to variants are explicitly set to either
ON/OFF.

The ITK_USE_MKL flag will turn on the following:
- USE_FFTWD=ON
- USE_FFTWF=ON
- USE_SYSTEM_FFTW=ON

Since the package depends on fftw-api, those options will always be set.

glennpj commented:
> It might be a good idea to explicitly disable some of these CMake flags in else clauses.

Agreed; I just pushed out a commit for that. That was an oversight on my part.

Thanks.
